pipes/rules/assembly.rules

```diff
@@ -103,7 +105,8 @@ rule orient_and_impute:
                 ono_extra.extend(["--min_pct_contig_aligned", str(config["scaffold_min_pct_contig_aligned"])])
             ono_extra = " ".join(ono_extra)
             shell("{config[bin_dir]}/assembly.py order_and_orient {input[0]} {params.refGenome} {params.scaffolded_fasta} {ono_extra} --outAlternateContigs {params.alternate_fasta}")
-            shell("{config[bin_dir]}/assembly.py impute_from_reference {params.scaffolded_fasta} {params.refGenome} {output} --newName {params.renamed_prefix}{wildcards.sample} --replaceLength {params.replace_length} --minLengthFraction {params.length} --minUnambig {params.min_unambig} --index")
+            shell("{config[bin_dir]}/assembly.py gapfill_gap2seq {params.scaffolded_fasta} {input.cleaned_reads} {params.gapfilled_fasta} --memLimitGb {resources[mem]} --maskErrors --randomSeed {config[random_seed]}")
```

Comment:
can we do the idiom here where we make a snakemake param for this step called `random_seed=config.get('random_seed', 0)` or something like that, and then use the param in the `shell` invocation?

Reply:
Setting random_seed to 0 in pipes/config.yaml would do that, no?

Reply:
The difference is that the current implementation fails on someone's older / pre-existing config.yaml until they hand merge in changes from the latest one.. it'd be nice to fallback to a sensible default if not specified.

Reply:
Also there are some rare scenarios where I've found it useful to specify param values on the command line when I invoked snakemake (which then overrides the config).. but I forget what situations I found that useful in... but that would require exposing it as a `param:` in the rule.

Reply:
you can change config[] values on the snakemake command line.  and with the config file inclusion PR, a legacy config file can omit new param values that are in the updated pipes/config.yaml, which you would get together with the updated pipes/rules that use the new param.